ios/PurchasesHybridCommon/PurchasesHybridCommon/SKProduct+HybridAdditions.m

```diff
@@ -6,27 +6,33 @@
 #import "SKProduct+HybridAdditions.h"
 #import "SKProductDiscount+HybridAdditions.h"
 
-@implementation SKProduct (RCPurchases)
+@import RevenueCat;
+
+@implementation RCStoreProduct (RCPurchases)
 
 - (nullable NSString *)rc_currencyCode {
     if(@available(iOS 10.0, tvOS 10.0, macOS 10.12, *)) {
-        return self.priceLocale.currencyCode;
+        // JOSH: this feels bad to use sk1 product
+        return self.sk1Product.priceLocale.currencyCode;
     } else {
-        return [self.priceLocale objectForKey:NSLocaleCurrencyCode];
+        // JOSH: this feels bad to use sk1 product
+        return [self.sk1Product.priceLocale objectForKey:NSLocaleCurrencyCode];
     }
 }
 
 - (NSDictionary *)rc_dictionary
 {
     NSNumberFormatter *formatter = [[NSNumberFormatter alloc] init];
     formatter.numberStyle = NSNumberFormatterCurrencyStyle;
-    formatter.locale = self.priceLocale;
+    // JOSH: this feels bad to use sk1 product
+    formatter.locale = self.sk1Product.priceLocale;
     NSMutableDictionary *d = [NSMutableDictionary dictionaryWithDictionary:@{
                         @"identifier": self.productIdentifier ?: @"",
                         @"description": self.localizedDescription ?: @"",
                         @"title": self.localizedTitle ?: @"",
-                        @"price": @(self.price.floatValue),
-                        @"price_string": [formatter stringFromNumber:self.price],
+                        // JOSH: sk1 used for now because Decimal is not getting bridged over from swift properly
+                        @"price": @([self.sk1Product.price floatValue]),
```

Comment:
Need to drop into `sk1Product` here because `Decimal` price on `StoreProduct` does not bridge to Objective-C correctly (https://forums.swift.org/t/decimal-imported-as-nsdecimal-not-nsdecimalnumber-in-swift-3-to-objective-c/4498/5)